include/cutlass/arch/mma_sm70.h

```diff
@@ -37,13 +37,13 @@
 #include "cutlass/layout/matrix.h"
 #include "cutlass/numeric_types.h"
 
-#if ((__CUDACC_VER_MAJOR__ > 10) || (__CUDACC_VER_MAJOR__ == 10 && __CUDACC_VER_MINOR__ >= 1))
+#if (defined(__clang__) || ((__CUDACC_VER_MAJOR__ > 10) || (__CUDACC_VER_MAJOR__ == 10 && __CUDACC_VER_MINOR__ >= 1)))
```

Comment:
Clang's clause should probably be a bit more detailed:
 `(defined(__clang__) && defined(__CUDA__) && CUDA_VERSION >= 10011)`

- `__CUDA__` will only be defined if clang is compiling CUDA sources. It will not be set when clang is used by NVCC during host-side compilation. It may not be necessary if mma_sm70.h is guaranteed to never be included from a C++ compilation.

- `CUDA_VERSION` is needed to make sure we enable new MMA instructions when clang uses sufficiently recent CUDA version. While nvcc version definitions and CUDA version are in sync, CUDACC_VER_* can be used to infer the CUDA version.. clang must use the `CUDA_VERSION` to do the same.

Reply:
Updated below.